File: Assets/Script/CZL/LuaCallStatic.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;


namespace LuaNameSpace
{
    public static class LuaStaticClass
    {
        public static int id = 10;

        public static string name
        {
            get;
            set;
        }

        public static string outPut()
        {
            return "static use";
        }

        public static void Defualt(string str="abc") 
        {
            Debug.Log(str);
        }
    }
}

public class LuaCallStatic : MonoBehaviour
{
    // Start is called before the first frame update
    //Assets/Script/CZL
    void Start()
    {
        
        XLuasEnv.Instance.LuaDoStrings("CZL/LuaCallStatic");
        //XLuasEnv.Instance.LuaDoStrings("CZL/LuaCallStatic");
        //XLuasEnv.Instance.LuaDoString("Assets/Script/CZL/LuaCallStatic1");
        XLuasEnv.Instance.LuaDoString_by_Path_FileName("DataPath/Lua/CZL", "LuaCallStatic12"); 
    }

    // Update is called once per frame
    private void OnDestroy()
    {
        XLuasEnv.Instance.FreeEnv();
    }
}

File: DataPath/Lua/CZL/LuaCallStatic.lua
--lua调用静态类
--规则""CS.命名空间.类名。成员变量名"
print("hello")
print("打印id号："..CS.LuaNameSpace.LuaStaticClass.id)

--静态属性赋值?
CS.LuaNameSpace.LuaStaticClass.name="admin"
print("名字:"..CS.LuaNameSpace.LuaStaticClass.name)

--静态成员方法?
print("静态成员方法?:"..CS.LuaNameSpace.LuaStaticClass.outPut())

--使用默认函数?
CS.LuaNameSpace.LuaStaticClass.Defualt()
--使用自定义函数?
local strs="iwanna"
CS.LuaNameSpace.LuaStaticClass.Defualt("自定义函数?:"..strs)